typesystem/struct.h

```diff
@@ -440,19 +440,33 @@ struct CurrentStructFieldsConsistency<T, 0u> {
   constexpr static const char* CURRENT_EXPORTED_STRUCT_NAME() { return #original_struct_name "_Z"; } \
   using template_inner_t_internal = std::true_type;                                                  \
   using template_inner_t_impl = original_template_inner_type
+
+#define CURRENT_USE_BASE_CONSTRUCTORS(s)                                                                        \
+  using CRNT_super_t =                                                                                          \
+      typename std::conditional<std::is_same<INSTANTIATION_TYPE, ::crnt::r::DF>::value, SUPER, CSSH_##s>::type; \
+  using CRNT_super_t::CRNT_super_t
+
+#define CURRENT_USE_T_BASE_CONSTRUCTORS(s)                                                               \
+  using CRNT_super_t = typename std::conditional<std::is_same<INSTANTIATION_TYPE, ::crnt::r::DF>::value, \
+                                                 SUPER,                                                  \
+                                                 CURRENT_STRUCT_T_SUPER_HELPER_##s>::type;               \
+  using CRNT_super_t::CRNT_super_t
 #else
 // I sure hope this is how it should be. -- D.K.
 #define CURRENT_EXPORTED_TEMPLATED_STRUCT(original_struct_name, original_template_inner_type)        \
   constexpr static const char* CURRENT_EXPORTED_STRUCT_NAME() { return #original_struct_name "_Z"; } \
   using template_inner_t = original_template_inner_type
+
+#define CURRENT_USE_BASE_CONSTRUCTORS(s) using SUPER::SUPER
```

Comment:
A comment explaining why this "simple" solution doesn't work on other platforms may help.